digitgen/digit_generator.py

```diff
@@ -0,0 +1,64 @@
+import os
+from time import process_time
+
+import numpy as np
+
+from .digit import DigitConfig, DigitSequence
+from .utils import CONFIGURATION
+from .utils import (format_annotations, generate_random_digits,
+                    test_annotations)
+
+
+class DigitGenerator(object):
+    def __init__(self, digit_size: int,
+                 font_file: str,
+                 samples=1,
+                 image_size= None,
+                 allowed_digits=["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"]) -> None:
+        self.digit_size = digit_size
+        self.samples = samples
+        self.image_size = image_size
+        self.allowed_digits = allowed_digits
+
+        self.config = CONFIGURATION
+        self.config["common_configs"]["font_file_loc"] = font_file
+
+        self.memory = {}  # dict.fromkeys(allowed_digits,None)
+
+    def generate(self, space_type="None", num_spaces=None, sectors=None, spaces_per_sector=None):
+        """
+        Generator function of the dataset
+
+        Returns:
+            Sequence[tuple[np.array,dict]]: img_array,annotation
+        """
+        # print(np.unique(random_array,return_counts=True))
```

Comment:
Another comment here.

Reply:
Removed the comment It was added during development